src/trpc/router/restRouter.ts

```diff
@@ -176,6 +206,265 @@ export const restRouter = router({
             return userMethods.edit({ id: user.id, ...input.data });
         }),
 
+    editUserByLogin: restProcedure
+        .meta({
+            openapi: {
+                method: 'PUT',
+                path: '/users/{login}',
+                protect: true,
+                summary: 'Update user by login',
+            },
+        })
+        .input(
+            z.object({
+                login: z.string(),
+                data: z.object({
+                    surname: z.string(),
+                    firstName: z.string(),
+                    middleName: z.string().optional(),
```

Comment:
why only middle name optional in update?

Reply:
Some people have only first name and surname, it's ok

Reply:
I mean, why all other fields is mandatory? What if we want to update just the phone number?

Reply:
Agree, all fields should be optional then. Besides, we can't update just name or surname, it's stored in db as single string